File: SistemaAleitamentoMaternoApi/SistemaAleitamentoMaternoApi/Dtos/LeiteMaternoDto.cs
using SistemaAleitamentoMaternoApi.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SistemaAleitamentoMaternoApi.Dtos
{
    public class LeiteMaternoDto : BaseDto
    {
        public bool Disponivel { get; set; } = true;
        [Required(ErrorMessage = "Você deve informar a pessoa que está realizando a doação.")]
        public Guid DoadorId { get; set; }
        [ForeignKey("BancoAleitamento")]
        public Guid? BancoAleitamentoId { get; set; }
        public Guid? ReceptorId { get; set; } = null;
        public DateTime? DataEntrada { get; set; } = DateTime.UtcNow;
        public DateTime? DataRetirada { get; set; } = DateTime.UtcNow;
    }
}

File: SistemaAleitamentoUI/src/app/models/leiteMaternoDTO.ts
import { BancoDTO } from 'src/app/models/bancoDTO';
export interface LeiteMaternoDTO {
  id: string;
  disponivel: boolean;
  doadorId: string
  bancoAleitamentoId: string;
  bancoAleitamento?: BancoDTO;
  receptorId?: string;
  dataEntrada?: string;
  dataRetirada?: string;
}
